src/mgmt/srv_pool.c

```diff
@@ -588,32 +587,49 @@ ds_mgmt_destroy_pool(uuid_t pool_uuid, const char *group, uint32_t force)
 	if (rc != 0) {
 		if (rc == -DER_ALREADY)
 			rc = 0;
+		goto out_ranks;
+	}
+
+	rc = pool_get_svc_ranks(svc, pool_uuid, &psvcranks);
+	if (rc != 0) {
+		D_ERROR("Failed to get pool service ranks "DF_UUID" rc: %d\n",
+			DP_UUID(pool_uuid), rc);
 		goto out_svc;
 	}
 
+	/* Check active pool connections, evict only if force */
+	rc = ds_pool_svc_check_evict(pool_uuid, psvcranks, force);
+	if (rc != 0) {
+		D_ERROR("Failed to check/evict pool handles "DF_UUID" rc: %d\n",
+			DP_UUID(pool_uuid), rc);
+		goto out_ranks;
+	}
+
 	rc = ds_pool_svc_destroy(pool_uuid);
```

Comment:
Is the question referring to current master (as opposed to what this patch adds)? There are two main components of ds_pool_svc_destroy():
ds_rebuild_leader_stop() - I am guessing the question is to do with this function invocation?
and ds_rsvc_dist_stop() - this seems to be OK to launch from the management service (rank 0) since the broadcast will reach the pool service leader.

Reply:
oh, I am asking if the whole destroy process should be driven by the pool leader. i.e. call ds_pool_svc_destroy in pool leader, otherwise ds_rebuild_leader_stop will work correctly, which has to be called in the pool leader. (Sorry, it is a bug in the current master).

I had thought this patch suppose to resolve this issue.

Reply:
A goal of this patch is to make sure when a pool is destroyed forcibly that there are no references remaining on the storage (due to existing pool connections) that would result in the storage not being fully reclaimed. It is a server-side solution that is an alternative to the client-side evict+destroy solution that was considered in PR-1241.

Reply:
ah, ok